### PR TITLE
Added guardrails profile to specs

### PR DESCRIPTION
**Pull request by rsamigullin**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [x] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2022-06-01/managedClusters.json [hunk 1]
```diff
@@ -5750,6 +5750,55 @@
         ]
       },
       "description": "The type of a snapshot. The default is NodePool."
+    },
+    "GuardrailsProfile": {
```
Comment (matthchr) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2022-06-01/managedClusters.json:5754]: So I see that this change was discussed and approved for `06-02-preveiw`, not `06-01` like you're targeting here. Also, the `06` APIs have already shipped, the Swagger cutoff was `07/03`.

Was your change in the RP but not in the Swagger? AFAIK this change should be targeted at the dev branch (which hasn't been created yet): https://github.com/Azure/azure-rest-api-specs/tree/dev-containerservice-Microsoft.ContainerService-2022-08-02-preview

@FumingZhang will know when it is planned to be created. Probably sometime around 8/19 as per the schedule.

**Review by matthchr — CHANGES_REQUESTED** [review]

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2022-06-01/managedClusters.json (hunk 1 above)
Comment (FumingZhang) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2022-06-01/managedClusters.json:5754]: Yes, I'll ping you in the api channel after the dev branch for 08-02-preview is created @rsamigullin

File: specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-08-02-preview/managedClusters.json [hunk 2]
```diff
@@ -6979,6 +7021,91 @@
...
+        },
+        "isPreview": {
+          "type": "boolean",
+          "description": "Describes whether this version is in preview"
+        },
+        "default": {
+          "type": "boolean",
+          "description": "Describes whether this version is default"
+        },
+        "data": {
```
Comment (FumingZhang) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-08-02-preview/managedClusters.json:7089]: From [CI](https://github.com/Azure/azure-rest-api-specs/pull/20171/checks?check_run_id=8025953973), missing identifier for this array.

May add`"x-ms-identifiers": [],` to bypass the check.

**Comment by FumingZhang** [comment]
The breaking changes detected in this PR has nothing to do with the changes in this PR, it is caused by this [PR #19592](https://github.com/Azure/azure-rest-api-specs/pull/19592). Will fix the issue in another [PR #20402](https://github.com/Azure/azure-rest-api-specs/pull/20402).

File: specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-08-02-preview/managedClusters.json [hunk 3]
```diff
@@ -6979,6 +7021,72 @@
...
+          "enum": [
+            "Off",
+            "Warning",
+            "Enforcement"
+          ],
+          "x-ms-enum": {
+            "name": "level",
+            "modelAsString": true
+          },
+          "description": "The guardrails level to be used"
```
Comment (matthchr) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-08-02-preview/managedClusters.json:7051]: Maybe also include here: "By default, Guardrails is enabled for all namespaces except those that AKS excludes via systemExcludedNamespaces"?

File: specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-08-02-preview/managedClusters.json [hunk 4]
```diff
@@ -6979,6 +7021,72 @@
...
+        },
+        "includedNamespaces": {
+          "description": "Customer-specified list of namespaces that must be included in guardrails",
```
[the remark this answers is not in the document]
Comment (matthchr) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-08-02-preview/managedClusters.json:7054]: Maybe also include here something like: "These namespaces cannot have guardrails turned off via the "aks-guardrails-silent:true" label"?

File: specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-08-02-preview/managedClusters.json [hunk 5]
```diff
@@ -6979,6 +7021,72 @@
...
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "excludedNamespaces": {
+          "description": "Customer-specified list of namespaces that must be excluded from guardrails",
```
Comment (matthchr) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-08-02-preview/managedClusters.json:7061]: Similar to above, consider phrasing this like:
- "List of namespaces excluded from guardrails" or 
- "List of namespaces excluded from guardrails checks" or

**Review by matthchr — APPROVED** [review]
Approved, but please look at and improve the descriptions some (see my suggestions for ideas)

File: specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-08-02-preview/managedClusters.json [hunk 6]
```diff
@@ -2891,6 +2891,48 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerService/managedClusters/{resourceName}/guardrailsversions": {
```
[the remark this answers is not in the document]
Comment (rsamigullin) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-08-02-preview/managedClusters.json:2895]: yeah, it's correct. I found this : 

``` ListGuardrailsVersionsLiteral    = "{gv:(?i)guardrailsversions}" ```

File: specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-08-02-preview/managedClusters.json (hunk 3 above)
Comment (rsamigullin) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-08-02-preview/managedClusters.json:7051]: Added

File: specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-08-02-preview/managedClusters.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (rsamigullin) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-08-02-preview/managedClusters.json:7054]: Upon further reflection we realized we don't need the includedNamespaces field and we decided to remove it. Please comment here if you have any objections

File: specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-08-02-preview/managedClusters.json (hunk 5 above)
Comment (rsamigullin) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-08-02-preview/managedClusters.json]: Updated

**Pull request merged**